### PR TITLE
Fix for #364

### PR DESCRIPTION
**Pull request by mawatech**
This PR should fix issue #364

Used https://github.com/jenkinsci/groovy-postbuild-plugin/commit/4aee64901e7e8ea589e0beeffc8508f0b57f992d as reference how to fix it.

Short test on a local Jenkins instance showed no problems.

### CODE REVIEW

**Comment by dcendents** [comment]
@mawatech if I'm not mistaken the BadgeAction has been moved to a new plugin, so technically there is no need to have a dependency on the groovy-postbuild plugin

I suggest you change the dependency in the pom

```xml
        <dependency>
            <groupId>org.jenkins-ci.plugins</groupId>
            <artifactId>badge</artifactId>
            <version>1.2</version>
        </dependency>
```

File: build-monitor-plugin/pom.xml [hunk 1]
```diff
@@ -166,7 +166,7 @@
         <dependency>
             <groupId>org.jvnet.hudson.plugins</groupId>
             <artifactId>groovy-postbuild</artifactId>
-            <version>2.3.1</version>
+            <version>2.4</version>
             <optional>true</optional>
         </dependency>
```
Comment (ikedam) [inline build-monitor-plugin/pom.xml]: +1 for the suggestion from @dcendents  .
This should be replaced with:
```
        <dependency>
            <groupId>org.jenkins-ci.plugins</groupId>
            <artifactId>badge</artifactId>
            <version>1.2</version>
            <optional>true</optional>
        </dependency>
```
(adding `optional=true` to the dcendents's comment)

And the plugin name to test should be also replaced to `badge`:
https://github.com/jan-molak/jenkins-build-monitor-plugin/blob/v1.11%2Bbuild.201701152243/build-monitor-plugin/src/main/java/com/smartcodeltd/jenkinsci/plugins/buildmonitor/viewmodel/JobViews.java#L18

**Comment by ikedam** [comment]
Badge plugin (and groovy-postbuild-2.4+) targets Jenkins 2.60.3 and acceptance test may not work correct. (I don't have an appropriate environment and haven't tried it...)
It may requires also upgrading the target version of build-monitor to 2.60.3.

**Comment by jan-molak** [comment]
@mawatech, @ikedam - thanks, I'll have a look at this PR now

**Comment by jan-molak** [comment]
Looks like the badge-related acceptance tests (`mvn clean verify`) are now failing:

scaled_2116ea922ddf9e7843f32338800b578b

**Comment by jan-molak** [comment]
OK, so upgrading to Jenkins 2.x looks like a fairly large piece of work. I'll have a look into that, but can't really promise any timeframes. Any help would be appreciated.

**Comment by ikedam** [comment]
I’m afraid upgrading the core version takes some more time.

@jan-molak I’m thinking of releasing groovy-postbuild-2.4.1 in the following way. Would you comment for this idea?

* Introduce `GroovyPostBuildAction` as a subclass of `BadgeAction` of badge-plugin.
* It resolves the startup failure.
* It results `HasBadges` not work at all, as actual actions are no longer `GroovyPostBuildAction` but `BadgeAction` (`allDetailsOf()` always return an empty list).
* I believe this is not the best way at all, but is better for now.

**Comment by jan-molak** [comment]
@ikedam - I think you're right. I spent a couple of hours looking into upgrading the jenkins core dependency, but it's not a straightforward process as the maven verifier they now have in place requires upgrading a ton of other dependencies. 

I'm not that familiar with the internals of the `groovy-postbuild` plugin, I'm afraid, but if you'd like to try out to implement your suggestion I'll do my best to help out.

Jan

**Comment by erikhakansson** [comment]
I've merged this into my #370 and also implemented backwards compatibility. This PR alone will break compatibility with old Jenkins' using the old Groovy Postbuild plugin.

**Comment by jan-molak** [comment]
Closed by #370

**Pull request closed without merging**